File: BLE_Fingerprinting/tests/test_ble_fingerprint.py
import numpy as np

from ble_fingerprint import ble_fingerprint
from gfsk_modulate import gfsk_modulate


def test_ble_fingerprint_returns_vector_or_none():
    rng = np.random.default_rng(2)
    bits = rng.integers(0, 2, size=32)
    fs = 1e6
    signal = gfsk_modulate(bits, 500e3, fs)

    fingerprint, decoded_bits = ble_fingerprint(
        signal,
        snr=40,
        fs=fs,
        preamble_detect=0,
        interp_fac=1,
        n_partition=1,
    )

    assert decoded_bits.shape[0] == len(bits)
    if fingerprint is not None:
        assert fingerprint.shape == (25,)

File: BLE_Fingerprinting/tests_matlab/test_ble_fingerprint.m
% Manual test for BLE_Fingerprint
rng(2);
bits = randi([0 1], 1, 32);
fs = 1e6;
signal = gfsk_modulate(bits, 500e3, fs);

[fingerprint, decoded] = BLE_Fingerprint(signal, 40, fs, 0, 1, 1);

assert(numel(decoded) == numel(bits));
if ~isempty(fingerprint)
    assert(size(fingerprint, 2) == 25);
end

disp('test_ble_fingerprint passed');
